src/app/pages/send-tokens/hooks/use-send-form.ts

```diff
@@ -15,22 +15,27 @@ export function useSendAmountFieldActions() {
   const { setFieldValue, values } = useFormikContext<SendFormValues>();
   const { data: stacksBalances } = useCurrentStacksAccountAnchoredBalances();
   const pendingTxsBalance = useCurrentAccountMempoolTransactionsBalance();
-  const { balance, isStx, selectedAssetBalance } = useSelectedAssetBalance(values.assetId);
+  const { isStx, selectedAssetBalance } = useSelectedAssetBalance(values.assetId);
 
   const handleSetSendMax = useCallback(
     (fee: number | string) => {
-      if (!selectedAssetBalance || !balance) return;
+      if (!selectedAssetBalance) return;
       if (isStx && fee) {
         const stx = microStxToStx(
           stacksBalances?.stx.availableStx.amount.minus(pendingTxsBalance) || 0
         ).minus(fee);
         if (stx.isLessThanOrEqualTo(0)) return;
         return setFieldValue('amount', stx.toNumber());
-      } else {
-        if (balance) setFieldValue('amount', removeCommas(balance));
       }
+      setFieldValue('amount', convertAmountToBaseUnit(selectedAssetBalance.balance).toString());
```

Comment:
this is outside the `convertAmountToBaseUnit`. We pass `Money` to `convertAmountToBaseUnit` and turn the resulting number to a string

Reply:
<img width="866" alt="image" src="https://user-images.githubusercontent.com/1618764/207620736-36217528-eaaf-4ab5-a6d6-4d705d87442d.png">

Reply:
Yes, I know this 😆 ...I am asking why you are turning it into a string rather than a number to work with in generating the unsigned tx with the form values? Our type is: `amount: number | string;`

Reply:
Our fee type is the same `fee: number | string;` ...and AFTER using `convertAmountToBaseUnit`, I called `toNumber()` instead of `toString()`.

Reply:
Sorry I'm lost. `FormikContext.setFieldValue` only accepts a string, so I `.toString()` it.

Is your question relating to code elsewhere? What does the unsigned transaction logic have to do with this? And what does "our type" refer to?